### PR TITLE
operator: move to real protobuf

### PR DESCRIPTION
**Pull request by howardjohn**
Currently, operator relies on some hacks to "fixup" the generated protobuf message. This is to workaround what was seen as limit in protobuf capabilities, but everything we needed was actually possible with some works.

This is important because we are currently blocked from updating protobuf; the changes in recent protubuf versions made the hacks in fixup_structs untenable. Additionally, this makes us standard protobuf, so we can do normal protobuf things.

The changes here should not have any user facing impact. Our tests provide pretty comprehensive coverage of this change.

This change depends on https://github.com/istio/api/pull/2272. My plan is to get this approved with a hold, then merge the API pr, then merge this.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -8,6 +8,8 @@ exclude k8s.io/kubernetes v1.13.0
 // Client-go does not handle different versions of mergo due to some breaking changes - use the matching version
 replace github.com/imdario/mergo => github.com/imdario/mergo v0.3.5
 
+replace istio.io/api => github.com/howardjohn/api v0.0.0-20220310185844-9962667afd9d
```
Comment (jacob-delgado) [inline go.mod:11]: I don't think you wanted to check this in

Comment (howardjohn) [reply go.mod:11]: Yeah I had a hold to fix this -- want to let tests run

**Review by litong01 — APPROVED** [review]
John, this has been tried in the past, the amount of work was a bit intimidating, glad that you have it done. Thanks so much!

**Review by esnible — APPROVED** [review]
LGTM (istioctl changes)

**Review by jacob-delgado — APPROVED** [review]

**Review by nmittler — APPROVED** [review]

**Pull request merged**